src/cast_core/src/CrudActor.cpp

```diff
@@ -928,6 +930,85 @@ struct CommitTransactionOperation : public BaseOperation {
     metrics::Operation _operation;
 };
 
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: withTransaction
+ *      OperationCommand:
+ *        OperationsInTransaction:
+ *          - OperationName: insertOne
+ *            OperationCommand:
+ *              Document: {a: 100}
+ *          - OperationName: findOneAndReplace
+ *            OperationCommand:
+ *              Filter: {a: 100}
+ *              Replacement: {a: 30}
+ */
+
+struct WithTransactionOperation : public BaseOperation {
+    WithTransactionOperation(const Node& opNode,
+                             bool onSession,
+                             mongocxx::collection collection,
+                             metrics::Operation operation,
+                             PhaseContext& context,
+                             ActorId id)
+            : BaseOperation(context, opNode),
+              _onSession{onSession},
+              _collection{std::move(collection)},
+              _operation{operation} {
+        auto& opsInTxn = opNode["OperationsInTransaction"];
+        if (!opsInTxn.isSequence()) {
+            BOOST_THROW_EXCEPTION(InvalidConfigurationException(
+                                          "'WithTransaction' requires an 'OperationsInTransaction' node of sequence type."));
+        }
+        for (const auto& [k, txnOp] : opsInTxn) {
```

Comment:
(nit) `const auto&&`, I think forward-referencing in ranged for loops is idiomatic in genny.

Reply:
👍